### PR TITLE
Documentation for IsolationMode #533

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

**Comment by Kerooker** [comment]
image

**Comment by sksamuel** [comment]
Is that good or bad ? lol

File: doc/isolation_mode.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+a=0
+a=0
+b=1
+a=0
+c=1
+
+
+
+### InstancePerLeaf
```
Comment (Kerooker) [inline doc/isolation_mode.md]: I do believe this is the default for JUnit, isn't it? If it is, maybe including it here would be good for people that are moving step by step

File: doc/reference.md [hunk 2]
```diff
@@ -3,6 +3,9 @@ KotlinTest
 
 [<img src="https://img.shields.io/maven-central/v/io.kotlintest/kotlintest-core.svg?label=latest%20release"/>](http://search.maven.org/#search|ga|1|kotlintest) [![GitHub license](https://img.shields.io/github/license/kotlintest/kotlintest.svg)]()
 
+This version of the document is for version 3.2.
```
Comment (Kerooker) [inline doc/reference.md]:
```suggestion
This version of the document is for version 3.2+.
```

Or else we'll forget this very quickly

File: doc/reference_3.1.md [hunk 3]
```diff
@@ -0,0 +1,1049 @@
+KotlinTest
+==========
+
+[<img src="https://img.shields.io/maven-central/v/io.kotlintest/kotlintest-core.svg?label=latest%20release"/>](http://search.maven.org/#search|ga|1|kotlintest) [![GitHub license](https://img.shields.io/github/license/kotlintest/kotlintest.svg)]()
+
+This version of the document is aimed at version 3.1
```
Comment (Kerooker) [inline doc/reference_3.1.md]: How long should we keep 3.1 reference for?
And apart from this line, is this the only one that changed?

**Comment by Kerooker** [comment]
> Is that good or bad ? lol

Good haha

**Comment by sksamuel** [comment]
> > Is that good or bad ? lol
> 
> Good haha

Remember, I'm not a young hipster developer :)

File: doc/reference_3.1.md (hunk 3 above)
Comment (sksamuel) [reply doc/reference_3.1.md]: reference 3.1 is just a copy of the original reference.md with that line added.

File: doc/isolation_mode.md (hunk 1 above)
Comment (sksamuel) [reply doc/isolation_mode.md]: Well none of the mode are strictly equivalent to junit as junit doesn't allow nested contexts does it ?

Comment (sksamuel) [reply doc/isolation_mode.md]: I've added some extra.

**Comment by Kerooker** [comment]
> > > Is that good or bad ? lol
> > 
> > 
> > Good haha
> 
> Remember, I'm not a young hipster developer :)

How could you think that doggos are bad tho?

**Comment by sksamuel** [comment]
> > > > Is that good or bad ? lol
> > > 
> > > 
> > > Good haha
> > 
> > 
> > Remember, I'm not a young hipster developer :)
> 
> How could you think that doggos are bad tho?

I suspected it was good.

**Comment by Kerooker** [comment]
What is trevis complaining about tho?

**Comment by sksamuel** [comment]
Another commit I did that this branch was forked from. It's fixed now.

**Pull request merged**